File: src/java/org/elastos/hive/vault/files/HashInfo.java
package org.elastos.hive.vault.files;

import com.google.gson.annotations.SerializedName;

class HashInfo {
	@SerializedName("name")
	private String name;

	@SerializedName("algorithm")
	private String algorithm;

	@SerializedName("hash")
	private String hash;

	public String getName() {
		return name;
	}

	public String getAlgorithm() {
		return algorithm;
	}

	public String getHash() {
		return hash;
	}
}

File: src/service/files/hashinfo.ts
export class HashInfo {
	private name: string;
	private algorithm: string;
	private hash: string;

	setName(name: string) {
	    this.name = name;
	    return this;
    }

    setAlgorithm(name: string) {
        this.name = name;
        return this;
    }

    setHash(name: string) {
        this.name = name;
        return this;
    }

    getName() {
	    return this.name;
    }

    getAlgorithm() {
        return this.algorithm;
    }

    getHash() {
        return this.hash;
    }
}
